controller/orphan_controller.go

```diff
@@ -0,0 +1,314 @@
+
+	if oc.queue.NumRequeues(key) < maxRetries {
+		log.WithError(err).Warnf("Error syncing Longhorn orphan %v: %v", key, err)
+
+		oc.queue.AddRateLimited(key)
+		return
+	}
+
+	utilruntime.HandleError(err)
+	log.WithError(err).Warnf("Dropping Longhorn orphan %v out of the queue: %v", key, err)
+	oc.queue.Forget(key)
+}
+
+func (oc *OrphanController) syncOrphan(key string) (err error) {
+	defer func() {
+		err = errors.Wrapf(err, "%v: fail to sync orphan %v", oc.name, key)
+	}()
+
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+	if namespace != oc.namespace {
+		return nil
+	}
+	return oc.reconcile(name)
+}
+
+func (oc *OrphanController) reconcile(orphanName string) (err error) {
+	orphan, err := oc.ds.GetOrphan(orphanName)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		return nil
+	}
+
+	log := getLoggerForOrphan(oc.logger, orphan)
+
+	if !oc.isResponsibleFor(orphan) {
+		return nil
+	}
+
+	if orphan.Status.OwnerID != oc.controllerID {
+		orphan.Status.OwnerID = oc.controllerID
+		orphan, err = oc.ds.UpdateOrphanStatus(orphan)
+		if err != nil {
+			// we don't mind others coming first
+			if apierrors.IsConflict(errors.Cause(err)) {
+				return nil
+			}
+			return err
+		}
+		log.Infof("Orphan Controller %v picked up %v", oc.controllerID, orphan.Name)
+	}
+
+	if !orphan.DeletionTimestamp.IsZero() {
+		defer func() {
+			if err == nil {
+				err = oc.ds.RemoveFinalizerForOrphan(orphan)
+			}
+		}()
+
+		// Make sure if the orphan node and controller ID are same.
+		// If NO, just delete the orphan resource object.
+		if orphan.Spec.NodeID != oc.controllerID {
+			return nil
+		}
+
+		node, err := oc.ds.GetNode(orphan.Spec.NodeID)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				log.Debugf("Only delete the orphan resource object since node %v does not exist", node.Name)
+				return nil
+			}
+			return err
+		}
+
+		if node.Spec.EvictionRequested {
+			return nil
+		}
+
+		id := orphan.Spec.Parameters[longhorn.OrphanDiskFsid]
+		if disk, ok := node.Spec.Disks[id]; ok {
+			if disk.EvictionRequested {
+				return nil
+			}
+		} else {
+			log.Debugf("Only delete the orphan resource object since disk %v on node %v does not exist", id, node.Name)
+			return nil
+		}
+
+		if node.Status.DiskStatus[id].DiskUUID == orphan.Spec.Parameters[longhorn.OrphanDiskUUID] {
+			err := oc.deleteOrphanedData(orphan)
+			if err != nil && !apierrors.IsNotFound(err) {
+				orphan.Status.Conditions = types.SetCondition(orphan.Status.Conditions,
+					longhorn.OrphanConditionTypeDeletable, longhorn.ConditionStatusFalse, "", err.Error())
+				log.WithError(err).Errorf("error deleting orphan %v data", orphan.Name)
+				return err
+			}
+		}
+		return nil
+	}
+
+	existingOrphan := orphan.DeepCopy()
+
+	orphan.Status.Conditions = types.SetCondition(orphan.Status.Conditions,
+		longhorn.OrphanConditionTypeDeletable, longhorn.ConditionStatusTrue, "", "")
```

Comment:
What if `orphan.Status.OwnerID != orphan.Spec.NodeID` or `oc.controllerID != = orphan.Spec.NodeID`?

Or in other words, does `deletable` here mean that the data can be cleaned up?